src/main/java/io/nats/client/NatsThread.java

```diff
@@ -33,7 +29,7 @@ public NatsThread(Runnable runnable, String name) {
         super(runnable, name + "-" + created.incrementAndGet());
         setUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler() {
             public void uncaughtException(Thread t, Throwable e) {
-                logger.debug("UNCAUGHT in thread {}", t.getName(), e);
+               e.printStackTrace();
```

Comment:
fair. not sure of the right answer here esp. since this is being thrown on a new thread. i guess stick with this and we'll see if we get feedback, revisit as we improve/simplify the rest of the code.